--- old/gr.h	2023-12-08 08:24:03
+++ new/gr.h	2023-12-08 08:21:57
@@ -1,20 +1,5 @@
-#ifndef _GR_H_
-#define _GR_H_
-
-#ifdef _WIN32
-
-#define HAVE_BOOLEAN
-
-#include <windows.h> /* required for all Windows applications */
-#define DLLEXPORT __declspec(dllexport)
-#else
 #define DLLEXPORT
-#endif
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 typedef enum
 {
   GR_PROJECTION_DEFAULT,
@@ -374,11 +359,5 @@
 DLLEXPORT double gr_volume_interp_gauss(const data_point3d_t *, const void *, const point3d_t *, const point3d_t *);
 DLLEXPORT void gr_setmathfont(int font);
 DLLEXPORT void gr_inqmathfont(int *font);
 DLLEXPORT void gr_setclipregion(int region);
 DLLEXPORT void gr_inqclipregion(int *region);
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif
