--- old/gr.h
+++ new/gr.h
@@ -1,20 +1,5 @@
-#ifndef _GR_H_
-#define _GR_H_
-
-#ifdef _WIN32
-
-#define HAVE_BOOLEAN
-
-#include <windows.h> /* required for all Windows applications */
-#define DLLEXPORT __declspec(dllexport)
-#else
 #define DLLEXPORT
-#endif
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 typedef enum
 {
   GR_PROJECTION_DEFAULT,
@@ -457,6 +442,0 @@
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif
