--- old/gr.h
+++ new/gr.h
@@ -1,20 +1,5 @@
-#ifndef _GR_H_
-#define _GR_H_
-
-#ifdef _WIN32
-
-#define HAVE_BOOLEAN
-
-#include <windows.h> /* required for all Windows applications */
-#define DLLEXPORT __declspec(dllexport)
-#else
 #define DLLEXPORT
-#endif
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 typedef enum
 {
   GR_PROJECTION_DEFAULT,
@@ -529,9 +514,3 @@
 DLLEXPORT char *gr_ftoa(char *string, double value, format_reference_t *reference);
 DLLEXPORT void gr_getformat(format_reference_t *result, double origin, double min, double max, double tick_width,
                             int major);
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif
